src/main/java/com/github/onsdigital/babbage/api/util/SearchRendering.java

```diff
@@ -0,0 +1,84 @@
+import static com.github.onsdigital.babbage.util.URIUtil.isDataRequest;
+
+/**
+ * SearchRendering contains common methods used to render a list of search results into a BabbageResponse.
+ * These were originally part of the SearchUtils package but have now been decoupled.
+ */
+public class SearchRendering {
+    private static final String ERRORS_KEY = "errors";
+
+    //Send result back to client
+    public static BabbageResponse buildResponse(HttpServletRequest request, String listType, Map<String, SearchResult> results) throws IOException {
+        if (isDataRequest(request.getRequestURI())) {
+            return buildDataResponse(listType, results);
+        } else {
+            return buildPageResponse(listType, results);
+        }
+    }
+
+    public static BabbageResponse buildDataResponse(String listType, Map<String, SearchResult> results) {
+        LinkedHashMap<String, Object> data = buildResults(listType, results);
+        return new BabbageStringResponse(JsonUtil.toJson(data), MediaType.APPLICATION_JSON, appConfig().babbage()
+                .getSearchResponseCacheTime());
+    }
+
+    public static BabbageResponse buildPageResponse(String listType, Map<String, SearchResult> results) throws IOException {
+        LinkedHashMap<String, Object> data = buildResults(listType, results);
+        return new BabbageStringResponse(TemplateService.getInstance().renderContent(data), MediaType.TEXT_HTML,
+                appConfig().babbage().getSearchResponseCacheTime());
+    }
+
+    public static BabbageResponse buildPageResponseWithValidationErrors(
+            String
+                    listType, Map<String, SearchResult>
+                    results, Optional<List<ValidationError>> errors
+    ) throws IOException {
+        LinkedHashMap<String, Object> data = buildResults(listType, results);
+        if (errors.isPresent() && !errors.get().isEmpty()) {
+            data.put(ERRORS_KEY, errors.get());
+        }
+        return new BabbageStringResponse(TemplateService.getInstance().renderContent(data), MediaType.TEXT_HTML,
+                appConfig().babbage().getSearchResponseCacheTime());
+    }
+
+    public static LinkedHashMap<String, Object> buildResults(
+            String
+                    listType, Map<String, SearchResult> results
+    ) {
+        LinkedHashMap<String, Object> data = getBaseListTemplate(listType);
+        if (results != null) {
+            for (Map.Entry<String, SearchResult> result : results.entrySet()) {
+                data.put(result.getKey(), result.getValue());
+            }
+        }
+        return data;
+    }
+
+    private static LinkedHashMap<String, Object> getBaseListTemplate(String listType) {
+        LinkedHashMap<String, Object> baseData = new LinkedHashMap<>();
+        baseData.put("type", "list");
+        baseData.put("listType", listType.toLowerCase());
+        baseData.put("uri", ((RequestUtil.Location) ThreadContext.getData("location")).getPathname());
```

Comment:
```
ThreadContext.getData("location")
```
Could return null which would cause an exception when `.getPathname()` is called. Can we add a not null defensive check around this step please?